### PR TITLE
Better connect disconnect for java script client api

### PR DESCRIPTION
**Pull request by kunsingh**

### CODE REVIEW

File: src/clientContextFactory.es6 [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        var beanManager = new BeanManager(classRepository);
+        var connector = new Connector(url, dolphin, classRepository, config);
+        var controllerManager = new ControllerManager(dolphin, classRepository, connector);
+
+        var clientContext = new ClientContext(dolphin, beanManager, controllerManager, connector);
+
+        return clientContext;
+    }
+}
+Emitter(ClientContextFactory.prototype);
```
Comment (hendrikebbers) [inline src/clientContextFactory.es6]: What does this does?

File: src/clientcontext.es6 [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        this.dolphin = dolphin;
+        this.beanManager = beanManager;
+        this._controllerManager = controllerManager;
+        this._connector = connector;
+        this.connectionPromise = null;
+        this.isConnected = false;
+    }
+
+    connect(){
+        var that = this;
```
Comment (hendrikebbers) [inline src/clientcontext.es6]: we should define if we need this helper var "self" or "that"

File: src/clientcontext.es6 [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    createController(name){
+        checkMethod('ClientContext.createController(name)');
+        checkParam(name, 'name');
+
+        return this._controllerManager.createController(name);
+    }
+
+    disconnect(){
+        // TODO: Implement ClientContext.disconnect [DP-46]
```
Comment (hendrikebbers) [inline src/clientcontext.es6]: Can we remove this TODO?

**Comment by timo0** [comment]
@hendrikebbers

**Pull request merged**